CONTRIBUTING.MD

```diff
@@ -0,0 +1,135 @@
+
+The actual architecture has only one [controller](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg#hdr-Controller) that's responsible to deploy the application from source. It's on the [roadmap](https://github.com/kiegroup/kogito-runtimes/wiki/Roadmap) to have two more controllers to handle SSO and Persistence. The following image illustrates the general idea:
+
+![Kogito Operator General Architecture](docs/img/general_archictecture.png?raw=true)
+
+One of the most important responsibilities of the controller is the [Reconcile Loop](https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md#reconcile-loop). Inside this "loop" the controller will ensure that it has every resource (Kubernetes and OpenShift objects) created and updated accordingly.
+
+We aim to avoid having a huge monolithic inside the reconcile loop that does it all. With that in mind, we separated the responsibility of the Kubernetes and OpenShift API calls to a package that we call [`inventory`](pkg/controller/kogitoapp/inventory) and the definition of the resources (our domain) into the [`definitions`](pkg/controller/kogitoapp/definitions) package.
+
+Take a look at the following diagram to have a general idea of what we're talking about:
+
+![Kogito Operator Packages Structure](docs/img/packages_structure.png?raw=true)
+
+`Controller` will orchestrate all operations through inventory calls by using its domain model (`Type`). `Inventory` will look for resources `definitions` once its needs to create new resources into the cluster.
+
+### Inventory
+
+In this package we handle all Kubernetes/OpenShift API calls, transforming those operations into meaningful functions that can be used across all controller operations. Take for example the `CreateResourceIfNotExists` function:
+
+```go
+func CreateResourceIfNotExists(cli client.Client, resource v1alpha1.OpenShiftObject) (bool, error) {
+	if exists, err := FetchResource(cli, resource); err == nil && !exists {
+		err = cli.Create(context.TODO(), resource)
+		if err != nil {
+			return false, err
+		}
+		return true, nil
+	} else if err != nil {
+		return false, err
+	}
+
+	return false, nil
+}
+```
+
+It will fetch for a particular named resource (e.g. a `ServiceAccount`), and if it does not exist, the function will create a new one using the API. 
+
+We try to make our best to have a code easy to read even for those who are not familiar with the Go language.
+
+### Definitions
+
+In the `definitions` package is defined every resource structure and its dependencies according to the controller requirements. The following diagram illustrates the relationship between the OpenShift resources for deploying a Kogito Application through the `KogitoApp` controller:
```

Comment:
The word order is wrong here. The easiest fix is, I think:
The definitions package defines the structure and dependencies of every resource according to the controller requirements.